lib/shopify_transporter/exporters/magento/customer_exporter.rb

```diff
@@ -12,14 +12,21 @@ def initialize(store_id: nil, soap_client: nil, database_adapter: nil)
 
         def export
           $stderr.puts 'Starting export...'
-          base_customers.map do |customer|
-            $stderr.puts "Fetching customer: #{customer[:customer_id]}..."
-            customer.merge(address_list: customer_address_list(customer[:customer_id]))
+          puts '['
+          base_customers.each do |base_customer|
+            puts ',' unless base_customer == base_customers.first
+            $stderr.puts "Fetching customer: #{base_customer[:customer_id]}..."
+            print '  ' + JSON.pretty_generate(with_attributes(base_customer), object_nl: "\n  ")
           end.compact
+          print "\n]"
```

Comment:
Add an `ensure` above this last print - in case there is an error or exception, within the `base_customers` loop, the file will still be complete.

Reply:
Aha! Maybe I should be rescuing inside every iteration of the loop too. Good thinking.